### PR TITLE
PUB-2847 | Extract action buttons from queue

### PR DESCRIPTION
**Pull request by MayaUribe**
## Description

- Extract share again and send to mobile buttons to shared-components
- Extract calendar buttons from the main queue

## Context & Notes

JIRA card -> PUB-2847

## Note
This is the second step towards refactoring multiple queues.
* **Next step:** PUB-2848: Merge PostLists and QueueItems

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/queue/components/QueueItems/index.jsx [hunk 1]
```diff
@@ -143,50 +147,57 @@ const renderPost = ({
...
-      <div style={{ marginLeft: 'auto' }}>
-        <QueueButtonGroup
-          buttons={calendarBtns}
-          onClick={type => onCalendarClick(type)}
-        />
-      </div>
-    )}
-  </div>
-);
+  shouldRenderCalendarButtons,
```
Comment (anafilipadealmeida) [inline packages/queue/components/QueueItems/index.jsx:162]: I don't know if that's something planned for a next step @MayaUribe, but I'd see `isUserPaid` more as a condition inside `shouldRenderCalendarButtons` and not as much as a prop to be passed in the Header

File: packages/shared-components/index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ export BitlyClickNotification from './BitlyClickNotification';
 export BufferLoading from './BufferLoading';
 export BusinessTrialOrUpgradeCard from './BusinessTrialOrUpgradeCard';
 export Card from './Card';
+export CalendarButtons from './CalendarButtons';
```
Comment (anafilipadealmeida) [inline packages/shared-components/index.js:10]: This is not a big deal, but they seem to be in alphabetical order 🙂

File: packages/queue/components/QueueItems/index.jsx (hunk 1 above)
Comment (MayaUribe) [reply packages/queue/components/QueueItems/index.jsx:162]: Yep, that makes sense 🤔 I'll move it :)

**Comment by MayaUribe** [comment]
> Seems good to me @MayaUribe! Just a heads up it seems the cypress test is failing!

It seemed to be a connection issue 🤔 thank you for the heads-up!

**Pull request merged**